internal/services/network/route_server_resource.go

```diff
@@ -272,6 +292,9 @@ func resourceRouteServerRead(d *pluginsdk.ResourceData, meta interface{}) error
 			if props.AllowBranchToBranchTraffic != nil {
 				d.Set("branch_to_branch_traffic_enabled", props.AllowBranchToBranchTraffic)
 			}
+			if props.HubRoutingPreference != nil {
+				d.Set("hub_routing_preference", string(*props.HubRoutingPreference))
+			}
```

Comment:
This can be simplified to 
```suggestion
			d.Set("hub_routing_preference", pointer.From(props.HubRoutingPreference))
```